spring-webclient-mocking/src/test/java/com/baeldung/webclientmocking/service/EmployeeServiceMockWebServerTest.java

```diff
@@ -0,0 +1,124 @@
+package com.baeldung.webclientmocking.service;
+
+import com.baeldung.webclientmocking.domain.Employee;
+import com.baeldung.webclientmocking.enums.Role;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.Rule;
+import org.junit.jupiter.api.*;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class EmployeeServiceMockWebServerTest {
+
+    public  static MockWebServer mockBackEnd=null;
+    private EmployeeService employeeService;
+    ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void setUp() throws IOException {
+        mockBackEnd = new MockWebServer();
+        mockBackEnd.start();
+    }
+
+    @AfterAll
+    static void tearDown() throws IOException {
+        mockBackEnd.shutdown();
+    }
+
+    @BeforeEach
+    void initialize(){
+
+        final String baseUrl=String.format("http://localhost:%s", mockBackEnd.getPort());
+        employeeService = new EmployeeService(baseUrl);
+        objectMapper = new ObjectMapper();
```

Comment:
This is not really a per-test-instance initialization. The `ObjectMapper` could just as easily be `private static final ObjectMapper MAPPER = new ObjectMapper();` and thus fade into the background, rather than be something the reader has to interpret here.